cardano-cli/src/Cardano/CLI/Shelley/Run/Pretty.hs

```diff
@@ -1,107 +1,162 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DefaultSignatures #-}
 {-# LANGUAGE DisambiguateRecordFields #-}
+{-# LANGUAGE FlexibleInstances #-}
 {-# LANGUAGE GADTs #-}
+{-# LANGUAGE LambdaCase #-}
 {-# LANGUAGE NamedFieldPuns #-}
 {-# LANGUAGE ScopedTypeVariables #-}
 
-module Cardano.CLI.Shelley.Run.Pretty (prettyTx) where
+module Cardano.CLI.Shelley.Run.Pretty (friendlyLBS) where
 
-import           Cardano.Api
-                   (ShelleyBasedEra (ShelleyBasedEraAllegra, ShelleyBasedEraMary, ShelleyBasedEraShelley))
+import           Cardano.Api as Api
+                   (ShelleyBasedEra (ShelleyBasedEraAllegra, ShelleyBasedEraMary, ShelleyBasedEraShelley),
+                   TxBody)
 import           Cardano.Api.Byron (TxBody (ByronTxBody))
 import           Cardano.Api.Shelley (TxBody (ShelleyTxBody))
+import           Cardano.Binary (Annotated)
 import           Cardano.CLI.Helpers (textShow)
+import qualified Cardano.Chain.UTxO as Byron
+import           Cardano.Ledger.Shelley (ShelleyEra)
+import           Cardano.Ledger.ShelleyMA (MaryOrAllegra (Allegra, Mary), ShelleyMAEra)
 import qualified Cardano.Ledger.ShelleyMA.TxBody as ShelleyMA
 import           Cardano.Prelude
-import           Data.Aeson as JSON (Value, object, (.=))
+import           Data.Aeson as JSON (Object, ToJSON, Value (..), object, toJSON, (.=))
 import           Data.Aeson.Encode.Pretty (Config (confCompare), defConfig, encodePretty')
+import qualified Data.HashMap.Strict as HashMap
+import qualified Data.Vector as Vector
+import           Ouroboros.Consensus.Shelley.Protocol.Crypto (StandardCrypto)
 import qualified Shelley.Spec.Ledger.API as Shelley
 
-prettyTx :: TxBody era -> LByteString
-prettyTx body0 =
-  encodePretty' defConfig{confCompare = compare} $
-  object $
-  case body0 of
-    ByronTxBody tx -> ["era" .= ("Byron" :: Text), "tx" .= tx]
+-- | User-friendly text representation
+class FriendlyJson a where
```

Comment:
What is the justification for creating the `FriendlyJson` typeclass? I think it would make more sense to extend `prettyTx` to suit our needs before creating another typeclass.

Reply:
I had a fight with the compiler on type families and GADT matching, but after you pointed, I figured a way to make it without an extra class.

But there are many types in consideration here, and there will be more soon. Typeclasses are a safer way to dispatch on a type than a set of functions.

Reply:
> But there are many types in consideration here, and there will be more soon
When there are more types to consider we can revisit implementing a class. I think for now its OK to rely on existing `ToJSON` instances and `encodePretty` https://hackage.haskell.org/package/aeson-pretty-0.8.8/docs/Data-Aeson-Encode-Pretty.html

Reply:
Yes, OK.